File: apps/ios/QuranMeali/Models/Surah.swift
//
//  Surah.swift
//  QuranMeali
//
//  Shapes mirror the backend DTOs at
//  QuranCompanion.Application.Features.Surahs.Dtos.*
//

import Foundation

enum RevelationPlace: String, Decodable {
    case meccan = "Meccan"
    case medinan = "Medinan"
    case unknown = "Unknown"

    var turkishLabel: String {
        switch self {
        case .meccan: return "Mekkî"
        case .medinan: return "Medenî"
        case .unknown: return "—"
        }
    }
}

struct SurahListItem: Decodable, Identifiable, Hashable {
    let number: Int
    let nameArabic: String
    let nameTurkish: String
    let nameTransliteration: String
    let verseCount: Int
    let revelationPlace: RevelationPlace

    var id: Int { number }
}

struct SurahDetail: Decodable, Identifiable, Hashable {
    let number: Int
    let nameArabic: String
    let nameTurkish: String
    let nameTransliteration: String
    let verseCount: Int
    let revelationPlace: RevelationPlace
    let displayOrder: Int

    var id: Int { number }
}

File: apps/android/app/src/main/java/com/bbozkurtcagri/quranmeali/model/Surah.kt
package com.bbozkurtcagri.quranmeali.model

import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = false)
data class SurahListItem(
    val number: Int,
    val nameArabic: String,
    val nameTurkish: String,
    val nameTransliteration: String,
    val verseCount: Int,
    val revelationPlace: RevelationPlace
)

@JsonClass(generateAdapter = false)
data class SurahDetail(
    val number: Int,
    val nameArabic: String,
    val nameTurkish: String,
    val nameTransliteration: String,
    val verseCount: Int,
    val revelationPlace: RevelationPlace
)

// Backend enum'u "Meccan" / "Medinan" / "Unknown" string'leri olarak gönderiyor —
// UI'da "Mekkî" / "Medenî" gösterilir; harita aşağıda.
enum class RevelationPlace(val turkishLabel: String) {
    Meccan("Mekkî"),
    Medinan("Medenî"),
    Unknown("—")
}
